packages/auth/src/client/service.ts

```diff
@@ -91,56 +89,47 @@ async function verifySig(
   return crypto.verify(null, data, publicKey, Buffer.from(sig))
 }
 
-function validateRequiredJwkProperties(jwk: JWKWithRequired): boolean {
-  if (
-    jwk.kty !== 'OKP' ||
-    (jwk.use && jwk.use !== 'sig') ||
-    (jwk.key_ops &&
-      (!jwk.key_ops.includes('sign') || !jwk.key_ops.includes('verify'))) ||
-    jwk.alg !== 'EdDSA' ||
-    jwk.crv !== 'Ed25519'
-  ) {
-    return false
-  }
-
-  return true
-}
-
-async function validateClientWithRegistry(
+async function validateClient(
   deps: ServiceDependencies,
   clientInfo: ClientInfo
 ): Promise<boolean> {
-  const { config } = deps
+  if (!isClientInfo(clientInfo)) return false
+
   const { jwk } = clientInfo.key
-  const { keyRegistries } = config
 
-  // No key registries in list means no validation
-  if (keyRegistries.length === 0) return true
-  if (!validateRequiredJwkProperties(jwk)) return false
+  const { keys, ...clientDetails } = await getClientDetailsByKid(deps, jwk.kid)
 
-  const kidUrl = new URL(jwk.kid)
-  if (!keyRegistries.includes(kidUrl.origin)) return false
+  if (
+    clientInfo.display.name !== clientDetails.name ||
+    clientInfo.display.uri !== clientDetails.uri ||
+    jwk.clientId !== clientDetails.id
+  )
+    return false
 
-  const { keys, ...registryClientInfo } = await getRegistryDataByKid(
-    deps,
-    jwk.kid
+  const currentKey = keys[0]
```

Comment:
The `kid` should refer to a specific key now, so it doesn't need to account for an array of them here.